src/main/kotlin/uk/gov/justice/digital/hmpps/incentivesapi/service/PrisonerIepLevelReviewService.kt

```diff
@@ -45,21 +45,32 @@ class PrisonerIepLevelReviewService(
   private val authenticationFacade: AuthenticationFacade,
   private val clock: Clock,
   private val featureFlagsService: FeatureFlagsService,
+  private val offenderSearchService: OffenderSearchService,
 ) {
   companion object {
     val log: Logger = LoggerFactory.getLogger(this::class.java)
   }
 
-  private fun setNextReviewDate(iepSummary: IepSummary): IepSummary {
+  private suspend fun setNextReviewDate(iepSummary: IepSummary): IepSummary {
     if (iepSummary.iepDetails.isEmpty()) {
       throw IllegalArgumentException("Cannot set nextReviewDate without iepDetails for bookingId = ${iepSummary.bookingId}")
     }
 
-    iepSummary.nextReviewDate = NextReviewDateService().calculate(
+    // NOTE: `iepSummary.prisonerNumber` could be `null` (when data is coming from NOMIS)
+    val prisonerNumber = iepSummary.prisonerNumber ?: run {
```

Comment:
As mentioned in the comment [in the `IepSummary`class the `prisonerNumber` is optional](https://github.com/ministryofjustice/hmpps-incentives-api/blob/main/src/main/kotlin/uk/gov/justice/digital/hmpps/incentivesapi/dto/IepSummary.kt#L22) and could be blank.

`prisonerNumber` would be blank when the data is coming from NOMIS/Prison API, but it will be populated when it's coming from our database (as [the column is not NULL-able](https://github.com/ministryofjustice/hmpps-incentives-api/blob/main/src/main/resources/db/migration/V1_7__prisoner_iep_level.sql#L6)).

Once we stop getting reviews from Prison API we can probably make the `prisonerNumber` mandatory and remove this logic to make this additional request to get the `prisonerNumber`.

Reply:
might be worth saving `prisonerNumber` into `iepSummary` in case it's needed in other calls later?

Reply:
I see your point. Given this fallback logic is gonna disappear when we only return data from the incentives' DB I'm very tempted to follow the single responsibility principle and _not_ set anything else aside from the `nextReviewDate`. 
Unless you have strong feelings about this of course :)